backend/app/views/spree/admin/promotions/rules/_taxon.html.erb

```diff
@@ -1,7 +1,7 @@
 <div class="card-body">
   <div class="form-group taxons_rule_taxons">
-    <%= label_tag "#{param_prefix}_taxon_ids_string", Spree.t('taxon_rule.choose_taxons') %>
-    <%= select_tag "#{param_prefix}[taxon_ids_string]", nil, class: 'taxon_picker d-block w-100', id: :product_taxon_ids %>
+    <%= label_tag "#{param_prefix}_taxon_ids", Spree.t('taxon_rule.choose_taxons') %>
+    <%= select_tag "#{param_prefix}[taxon_ids]", options_from_collection_for_select(promotion_rule.taxons, :id, :pretty_name, promotion_rule.taxons.pluck(:id)), class: 'taxon_picker d-block w-100', multiple: true, id: :product_taxon_ids %>
```

Comment:
```suggestion
    <%= select_tag "#{param_prefix}[taxon_ids]", options_from_collection_for_select(promotion_rule.taxons, :id, :pretty_name, promotion_rule.taxon_ids), class: 'taxon_picker d-block w-100', multiple: true, id: :product_taxon_ids %>
```

Reply:
thank you! applied https://github.com/spree/spree/pull/10814/commits/444cd03c8883ff0e8567edfcdbe7afee99020ebc